### PR TITLE
[MAINTENANCE] Migrate ci to enterprise-arc runners

### PR DESCRIPTION
**Pull request by nicgrayson**

### CODE REVIEW

File: tasks.py [hunk 1]
```diff
@@ -1042,7 +1042,7 @@ def ci_tests(  # noqa: C901 - too complex (9)
     pytest_options = [f"--durations={slowest}", "-rEf"]
 
     if xdist:
-        pytest_options.append("-n auto")
+        pytest_options.append("-n 4")
```
Comment (billdirks) [inline tasks.py:1088]: Did you mean to hard code this to 4?

**Review by billdirks — APPROVED** [review]
Thanks! One question about the xdist number of workers.

File: tasks.py (hunk 1 above)
Comment (nicgrayson) [reply tasks.py:1088]: This was on purpose. Kubernetes exposes all cores of the host machine to the pod even if the pod is allocated fewer

Comment (Kilo59) [reply tasks.py:1088]: Will logical core give the right number?
`4` might be good for CI, but it'll slow things down when testing locally.

Can you leave this as `auto` (or `logical`) and update the the `ci.yaml` to pass the hardcoded 4 value.

Comment (nicgrayson) [reply tasks.py:1088]: no, should we make this something that changes when it detects that it's running in CI?

**Pull request merged**